### PR TITLE
Blog listing migrate

### PR DESCRIPTION
**Pull request by lowyuenan**

### CODE REVIEW

File: src/blog.tsx [hunk 1]
```diff
@@ -1,59 +1,112 @@
...
+    outputFolder: getConfig(state).outputFolder,
+  })
+}
+
+function renderPage(state: State) {
+  const page = getPages(state).find(it => it.id === Blog.id)!
+  const openGraph = getOpenGraph(state).find(it => it.pageId === Blog.id)
+  const currentAbsoluteUrl = `${state.config.protocol}://${join(state.config.hostname, page.url)}`
+  const blogPosts = getBlogPosts(state).sort(comparePublishedDate)
+  const allPages = state.website.pages
```
Comment (danielepolencic) [inline src/blog.tsx]: You have already `allPages` on line 52

Comment (lowyuenan) [reply src/blog.tsx]: But line 52 is only the blog page. `allPages` needed for all blog posts title and description.

Comment (danielepolencic) [reply src/blog.tsx]: You're right. `getBlogPosts` then

Comment (lowyuenan) [reply src/blog.tsx]: oh, I am Sorry, I am telling it wrong, `allPages` needed for all blog posts url, title and description are get from `getBlogPosts`.

Comment (lowyuenan) [reply src/blog.tsx]: I think this commit `1282369` will make it better.

**Comment by danielepolencic** [comment]
LGTM

**Pull request merged**